### ISSUE TITLE
lr_scheduler do not support ReduceLROnPlateau

### ISSUE THREAD

**Issue by zheLim** [NervanaSystems/distiller#161]
ReduceLROnPlateau takes an metric( such as validation loss) as input. But on LRPolicy.on_epoch_begin do not received metrics.

**Comment by nzmora** [comment]
Hi @zheLim, 

If you need to use ReduceLROnPlateau in the meanwhile, you can quickly add this in classifier_compressor.py manually.

Cheers
Neta

**Comment by guyjacob** [comment]
Support added with #182. It allows passing additional keyword arguments to `on_epoch_begin`. In the case of `LRPolicy` using `ReduceLROnPlateau`, if an argument named `metrics` was passed it'll be passed on the LR scheduler.
Note that the examples in the PyTorch docs show that ReduceLROnPlateau is called at the end of the epoch. We call it at the beginning, similarly to other LR schedulers, which means special handling is required on the first epoch. We might change this in the future.
For a usage example see the image classification sample:
https://github.com/NervanaSystems/distiller/blob/master/examples/classifier_compression/compress_classifier.py#L275-L277
